gofmt.go

```diff
@@ -130,6 +131,46 @@ func isGenerated(file *ast.File) bool {
 	return false
 }
 
+func shouldSkipPath(ignoredPaths []*modfile.Ignore, path string) bool {
+	base := filepath.Base(path)
+	// vendor and testdata directories are skipped,
+	// unless they are explicitly passed as an argument.
+	if base == "vendor" || base == "testdata" {
+		return true
+	}
+
+	// ignore paths that specified in go.mod's ignore directive
+	for _, i := range ignoredPaths {
+		if path == i.Path {
+			return true
+		}
+		if isSubPath(i.Path, path) {
```

Comment:
If we used `strings.Contains(path, i.Path)`, `shouldSkipPath` would incorrectly return `true` when `path` is `/foo/bar` and `i.Path` is `/foo/b`, because `/foo/bar` contains the substring `/foo/b` even though `/foo/bar` is not actually a child directory of `/foo/b`.

That's why `isSubPath` is necessary.